### PR TITLE
Propose a CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by Hwesta**
This proposes contributing guideline for FIDO.  Notably, it proposes a process for approving pull requests and creating new releases.  Please provide feedback!

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+* What did you expect to happen?
+* What did you see instead?
+* Can you reproduce this reliably?
+
+
+## Submitting code changes
+
+Every new feature and bugfix is expected to go through code review before inclusion.
+We use Github pull requests for code review.
+
```
Comment (Hwesta) [inline CONTRIBUTING.md]: This should be expanded to actually discuss code standards.  What code standards do we want, besides PEP8?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+## Code Review & Approval
+
+After a pull request is submitted, a maintainer will review the code for content, style & suitability of the new feature.
+After the reviewer is satisfied with the changes, they will approve it on Github.
+If at least two maintainers approve and there are no disapprovals for a week, the branch can be merged.
```
Comment (Hwesta) [inline CONTRIBUTING.md]: Is two maintainers too many, or should it be just one?  Is a week too long, or too short for everyone to have a chance to look at things?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+If the original committer is a maintainer, they can merged the branch.
+For external contributors, a maintainer wire merge the commits.
+
+
+## Release process
+
+A new release can happen when the maintainers decide enough changes have been merged, or when a new release of PRONOM happens.
```
Comment (Hwesta) [inline CONTRIBUTING.md]: Should a PRONOM release automatically trigger a FIDO release?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+To propose a release, a maintainer creates an issue with the release number and proposed release notes.
+Discussion can happen in the release ticket about what outstanding pull requests could be merged.
+If at least two maintainers approve and there are no disapprovals for a week, the release can be tagged and uploaded to PyPI.
```
Comment (Hwesta) [inline CONTRIBUTING.md]: Is this enough/not enough people? Enough/not enough time?

Comment (mistydemeo) [inline CONTRIBUTING.md]: Two maintainers is probably appropriate for a release.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+We use Github pull requests for code review.
+
+
+## Code Review & Approval
+
+After a pull request is submitted, a maintainer will review the code for content, style & suitability of the new feature.
+After the reviewer is satisfied with the changes, they will approve it on Github.
+If at least two maintainers approve and there are no disapprovals for a week, the branch can be merged.
+If the original committer is a maintainer, they can merged the branch.
+For external contributors, a maintainer wire merge the commits.
```
Comment (mistydemeo) [inline CONTRIBUTING.md]: `will merge`?

File: CONTRIBUTING.md (hunk 1 above)
Comment (mistydemeo) [inline CONTRIBUTING.md]: Hmm, good question. Is there another widely-accepted Python community standard we can borrow here?

Comment (Hwesta) [inline CONTRIBUTING.md]: Possibly PEP 257 for docstrings.  We might also want to discuss: Python 2 and/or 3 support, line length, semantic line breaks, requiring tests, commit message style.

**Comment by jrwdunham** [comment]
I am closing this in favour of the rebased https://github.com/openpreserve/fido/pull/127 which should also address the comments/change requests of @mistydemeo.

**Pull request closed without merging**